content/articles/working-with-coroutine-in-python/index.md

```diff
@@ -0,0 +1,197 @@
+2. Corporative coroutines build a pipeline structure while they are being executed.But while using subroutines, you may create an executable with a linear flow.
+3. Suspension and resumption are from a single point in subroutines while in coroutines is from multiple points. In coroutines, we can still resume execution from where we interrupted it.
+4. The main function shows the results of data processing in subroutines unlike in coroutines where the output of a coroutine may be seen.
+5. Python coroutines do not have a primary function to arrange and coordinate their execution. Subroutines have a main function for controlling and co-ordinating subroutine execution in python.
+
+### How different are coroutines from threads?
+With threads, the operating system changes running threads proactively as per its scheduler, which is an algorithm in the operating system kernel.
+
+Activities are collaboratively multi-tasked by stopping and restarting functions at predetermined times, it can sometimes be within a single thread. This is accomplished through the use of coroutines.
+
+### Coroutines in a Python program
+Coroutines will use the `send()` method to transmit names, and only those with the prefix "Student" will be displayed at the end.
+
+```python
+def print_cognomen (prfx):
+    print("The coroutine object is looking for the prefix: {}".format(prfx))
+    while True: 
+    GivenCognomen = (yield) 
+    if prfx in GivenCognomen:
+      print(GivenCognomen) 
+
+CorouteObject = print_cognomen("Student")
+
+cognomen_a = input("Fill 1st cognomen : ") 
+cognomen_b = input("Fill 2nd cognomen : ") 
+cognomen_c = input("Fill 3rd cognomen : ") 
+
+CorouteObject.__next__() 
+CorouteObject.send(cognomen_a) 
+CorouteObject.send(cognomen_b) 
+CorouteObject.send(cognomen_c)
+```
+
+#### Output
+
+```txt
+Fill 1st cognomen : 
+Johson 
+Fill 2nd cognomen : 
+Gitau
+Fill 3rd cognomen : 
+Ngure
+The coroutine object is looking for the prefix.: Student
+```
+
+#### Description
+The names were sent to the function's coroutine using the `send()` technique. As the output shows only the names that march the prefix `Student` has been reported. `Student` has been used as a coroutine keyword.
+
+### Coroutines execution
+Python coroutines and Python generators have a lot in common, as do many other types of Python code. Unless and until the coroutine is called in a program, nothing else happens.
+
+Execution begins when `__next__()` is invoked. The initial expression is performed when invoked.
+
+The coroutine's execution comes to a halt as it awaits the value to be sent to the coroutine object. Each time a new coroutine object receives an initial value, it first checks to see whether the specified prefix is present before printing the name with that prefix.
+
+For as long as you're using the this phrase, you'll keep seeing it until you run across the `name = (yield)` expression again.
+

```

Comment:
@Nganga89 Please clearly outline what you are referring to by "this" in `this phrase` and "it" in `seeing it`.